components/ai_chat/core/browser/engine/conversation_api_client_unittest.cc

```diff
@@ -603,6 +661,118 @@ TEST_F(ConversationAPIUnitTest, PerformRequest_NonPremium) {
   testing::Mock::VerifyAndClearExpectations(credential_manager_.get());
 }
 
+TEST_F(ConversationAPIUnitTest, PerformRequest_WithToolUseResponse) {
+  // Tests that we interpret tool use reponses. For more variants
+  // see tests for `ToolUseEventFromToolCallsResponse`.
+  std::vector<ConversationAPIClient::ConversationEvent> events =
+      GetMockEventsAndExpectedEventsBody().first;
+
+  MockAPIRequestHelper* mock_request_helper =
+      client_->GetMockAPIRequestHelper();
+  testing::StrictMock<MockCallbacks> mock_callbacks;
+  base::RunLoop run_loop;
+
+  // Intercept API Request Helper call and verify the request is as expected.
+  // Tool use is only supported for streaming requests since the completion
+  // callback only supports a single event.
+  EXPECT_CALL(*mock_request_helper, RequestSSE)
+      .WillOnce([&](const std::string& method, const GURL& url,
+                    const std::string& body, const std::string& content_type,
+                    DataReceivedCallback data_received_callback,
+                    ResultCallback result_callback,
+                    const base::flat_map<std::string, std::string>& headers,
+                    const api_request_helper::APIRequestOptions& options) {
+        {
+          base::Value result(base::Value::Type::DICT);
+          result.GetDict().Set("type", "completion");
+          result.GetDict().Set("model", "model-1");
+          result.GetDict().Set("completion", "This is a test completion");
+          result.GetDict().Set("tool_calls", base::test::ParseJsonList(R"([
+              {
+                "id": "call_123",
+                "type": "function",
+                "function": {
+                  "name": "get_weather",
+                  "arguments": "{\"location\":\"New York\"}"
```

Comment:
can we test streaming the arguments in for a tool?

Reply:
since this is just a string now I'm not sure how worth it to test the particular content of the string in this test. We'll have that in the place that we do the parsing